flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/dataformat/Timestamp.java

```diff
@@ -0,0 +1,167 @@
+		if (cmp == 0) {
+			cmp = this.nanoOfMillisecond - that.nanoOfMillisecond;
+		}
+		return cmp;
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (!(obj instanceof Timestamp)) {
+			return false;
+		}
+		Timestamp that = (Timestamp) obj;
+		return this.millisecond == that.millisecond &&
+				this.nanoOfMillisecond == that.nanoOfMillisecond;
+	}
+
+	@Override
+	public String toString() {
+		return toLocalDateTime().toString();
+	}
+
+	@Override
+	public int hashCode() {
+		int ret = (int) millisecond ^ (int) (millisecond >> 32);
+		return 31 * ret + nanoOfMillisecond;
+	}
+
+	public long getMillisecond() {
+		return millisecond;
+	}
+
+	public int getNanoOfMillisecond() {
+		return nanoOfMillisecond;
+	}
+
+	/**
+	 * Obtains an instance of {@code Timestamp} from a millisecond.
+	 *
+	 * <p>This returns a {@code Timestamp} with the specified millisecond.
+	 * The nanoOfMillisecond field will be set to zero.
+	 *
+	 * @param millisecond the number of milliseconds since January 1, 1970, 00:00:00 GMT
+	 *                    A negative number is the number of milliseconds before
+	 *                    January 1, 1970, 00:00:00 GMT
+	 * @return an instance of {@code Timestamp}
+	 */
+	public static Timestamp fromLong(long millisecond) {
+		return new Timestamp(millisecond, 0);
+	}
+
+	/**
+	 * Convert this {@code Timestmap} object to a {@link java.sql.Timestamp}.
+	 *
+	 * @return An instance of {@link java.sql.Timestamp}
+	 */
+	public java.sql.Timestamp toTimestamp() {
+		return java.sql.Timestamp.valueOf(toLocalDateTime());
+	}
+
+	/**
+	 * Obtains an instance of {@code Timestamp} from an instance of {@link java.sql.Timestamp}.
+	 *
+	 * <p>This returns a {@code Timestamp} with the specified {@link java.sql.Timestamp}.
+	 *
+	 * @param ts an instance of {@link java.sql.Timestamp}
+	 * @return an instance of {@code Timestamp}
+	 */
+	public static Timestamp fromTimestamp(java.sql.Timestamp ts) {
+		return fromLocalDateTime(ts.toLocalDateTime());
+	}
+
+	/**
+	 * Convert this {@code Timestamp} object to a {@link LocalDateTime}.
+	 *
+	 * @return An instance of {@link LocalDateTime}
+	 */
+	public LocalDateTime toLocalDateTime() {
+		int date = (int) (millisecond / MILLIS_PER_DAY);
+		int time = (int) (millisecond % MILLIS_PER_DAY);
+		if (time < 0) {
+			--date;
+			time += MILLIS_PER_DAY;
+		}
+		long nanoOfDay = time * 1_000_000L + nanoOfMillisecond;
+		LocalDate localDate = LocalDate.ofEpochDay(date);
+		LocalTime localTime = LocalTime.ofNanoOfDay(nanoOfDay);
+		return LocalDateTime.of(localDate, localTime);
+	}
+
+	/**
+	 * Obtains an instance of {@code Timestamp} from an instance of {@link LocalDateTime}.
+	 *
+	 * <p>This returns a {@code Timestamp} with the specified {@link LocalDateTime}.
+	 *
+	 * @param dateTime an instance of {@link LocalDateTime}
+	 * @return an instance of {@code Timestamp}
+	 */
+	public static Timestamp fromLocalDateTime(LocalDateTime dateTime) {
```

Comment:
You can implement it use `LocalDate.toEpochDay` and `LocalTime.toNanoOfDay`.
It is just a reversal of `toLocalDateTime`.

Reply:
OK. use this could remove calcite avatica function `unixTimestamp`